log-replication/src/main/java/org/corfudb/logreplication/runtime/GRPCLogReplicationClientChannelAdapter.java

```diff
@@ -0,0 +1,109 @@
+package org.corfudb.logreplication.runtime;
+
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.CustomClientRouter;
+import org.corfudb.infrastructure.IClientChannelAdapter;
+import org.corfudb.runtime.Messages.CorfuMessage;
+import org.corfudb.runtime.LogReplicationChannelGrpc;
+
+/**
+ * This is a default implementation of a custom channel for Log Replication Servers inter-communication
+ * which relies on the standard GRPC.
+ *
+ * It is used for testing purposes.
+ *
+ */
+@Slf4j
+public class GRPCLogReplicationClientChannelAdapter extends IClientChannelAdapter {
+
+    private final Channel channel;
+    private final LogReplicationChannelGrpc.LogReplicationChannelBlockingStub blockingStub;
+    private final LogReplicationChannelGrpc.LogReplicationChannelStub asyncStub;
+
+    private StreamObserver<CorfuMessage> requestObserver;
+    private StreamObserver<CorfuMessage> responseObserver;
+
+    /** Construct client for accessing LogReplicationService server using the existing channel. */
+    public GRPCLogReplicationClientChannelAdapter(Integer port, String host, CustomClientRouter adapter) {
+        super(port, host, adapter);
+        channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+        blockingStub = LogReplicationChannelGrpc.newBlockingStub(channel);
+        asyncStub = LogReplicationChannelGrpc.newStub(channel);
+    }
+
+    @Override
+    public void send(CorfuMessage msg) {
```

Comment:
would it be better to break the send into synchronous and asynchronous methods. Also maybe the name of the method should indicate that the response is observed on an observer.